.github/workflows/upgrade-testing.yml

```diff
@@ -0,0 +1,167 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'ubuntu-latest' ]
+        php: [ '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2' ]
+        db-type: [ 'mysql' ]
+        db-version: [ '5.7', '8.0' ]
+        wp: [ '6.0', '6.1', '6.2' ]
+        multisite: [ false, true ]
+    with:
+      os: ${{ matrix.os }}
+      php: ${{ matrix.php }}
+      db-type: ${{ matrix.db-type }}
+      db-version: ${{ matrix.db-version }}
+      wp: ${{ matrix.wp }}
+      new-version: ${{ inputs.new-version && inputs.new-version || '' }}
+      multisite: ${{ matrix.multisite }}
+
+  #
+  # Spawns upgrade testing for WordPress 5.x versions on PHP 7.x with MySQL.
+  #
+  upgrade-tests-wp-5x-php-7x-mysql:
+    name: ${{ matrix.wp }} to ${{ inputs.new-version && inputs.new-version || 'latest' }}
+    uses: WordPress/wordpress-upgrade-tester/.github/workflows/upgrade-testing-run.yml@combine-upgrade-test-workflows
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'ubuntu-latest' ]
+        php: [ '7.0', '7.1', '7.2', '7.3', '7.4' ]
+        db-type: [ 'mysql' ]
+        db-version: [ '5.7', '8.0' ]
+        wp: [ '5.0', '5.1', '5.2', '5.3', '5.4', '5.5', '5.6', '5.7', '5.8', '5.9' ]
+        multisite: [ false, true ]
+    with:
+      os: ${{ matrix.os }}
+      php: ${{ matrix.php }}
+      db-type: ${{ matrix.db-type }}
+      db-version: ${{ matrix.db-version }}
+      wp: ${{ matrix.wp }}
+      new-version: ${{ inputs.new-version && inputs.new-version || '' }}
+      multisite: ${{ matrix.multisite }}
+
+  #
+  # Spawns upgrade testing for WordPress 5.x versions on PHP 8.x with MySQL.
+  #
+  # WordPress 5.0-5.2 are excluded from PHP 8+ testing because of the following fatal errors:
+  # - Use of __autoload().
+  # - array/string offset with curly braces.
+  #
+  upgrade-tests-wp-5x-php-8x-mysql:
+    name: ${{ matrix.wp }} to ${{ inputs.new-version && inputs.new-version || 'latest' }}
+    uses: WordPress/wordpress-upgrade-tester/.github/workflows/upgrade-testing-run.yml@combine-upgrade-test-workflows
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'ubuntu-latest' ]
+        php: [ '8.0', '8.1', '8.2' ]
+        db-type: [ 'mysql' ]
+        db-version: [ '5.7', '8.0' ]
+        wp: [ '5.3', '5.4', '5.5', '5.6', '5.7', '5.8', '5.9' ]
+        multisite: [ false, true ]
+    with:
+      os: ${{ matrix.os }}
+      php: ${{ matrix.php }}
+      db-type: ${{ matrix.db-type }}
+      db-version: ${{ matrix.db-version }}
+      wp: ${{ matrix.wp }}
+      new-version: ${{ inputs.new-version && inputs.new-version || '' }}
+      multisite: ${{ matrix.multisite }}
+
+  #
+  # Spawns upgrade testing for WordPress 4.x versions on PHP 7.x with MySQL.
+  #
+  upgrade-tests-wp-4x-php-7x-mysql:
```

Comment:
Separating out the PHP 7.x and 8.x strategies:
- eliminates the need for any `exclude` or `include` statements.
- Ensures each group falls under the 256 maximum job limit for a strategy matrix.